bindings/wasm/src/tangle/resolver.rs

```diff
@@ -0,0 +1,231 @@
+        .map(WasmDocumentHistory::from)
+        .map(Into::into)
+        .wasm_result()
+    });
+
+    // WARNING: this does not validate the return type. Check carefully.
+    Ok(promise.unchecked_into::<PromiseDocumentHistory>())
+  }
+
+  /// Returns the `DiffChainHistory` of a diff chain starting from a `Document` on the
+  /// integration chain.
+  ///
+  /// NOTE: the document must have been published to the Tangle and have a valid message id.
+  #[wasm_bindgen(js_name = resolveDiffHistory)]
+  pub fn resolve_diff_history(&self, document: &WasmResolvedDocument) -> Result<PromiseDiffChainHistory> {
+    let resolved_document: ResolvedIotaDocument = document.0.clone();
+
+    let resolver: Rc<Resolver> = Rc::clone(&self.0);
+    let promise: Promise = future_to_promise(async move {
+      resolver
+        .resolve_diff_history(&resolved_document)
+        .await
+        .map(DiffChainHistory::from)
+        .map(Into::into)
+        .wasm_result()
+    });
+
+    // WARNING: this does not validate the return type. Check carefully.
+    Ok(promise.unchecked_into::<PromiseDiffChainHistory>())
+  }
+
+  /// Fetches the DID Document of the issuer on a `Credential`.
+  ///
+  /// # Errors
+  ///
+  /// Errors if the issuer URL is not a valid `DID` or document resolution fails.
+  #[wasm_bindgen(js_name = resolveCredentialIssuer)]
+  pub fn resolve_credential_issuer(&self, credential: &WasmCredential) -> Result<PromiseResolvedDocument> {
+    // TODO: reimplemented function to avoid cloning the entire credential.
+    //       Would be solved with Rc internal representation, pending memory leak discussions.
+    let issuer: IotaDID = IotaDID::parse(credential.0.issuer.url().as_str()).wasm_result()?;
+
+    let resolver: Rc<Resolver> = Rc::clone(&self.0);
+    let promise: Promise = future_to_promise(async move {
+      resolver
+        .resolve(&issuer)
+        .await
+        .map(WasmResolvedDocument::from)
+        .map(Into::into)
+        .wasm_result()
+    });
+
+    // WARNING: this does not validate the return type. Check carefully.
+    Ok(promise.unchecked_into::<PromiseResolvedDocument>())
+  }
+
+  /// Fetches all DID Documents of `Credential` issuers contained in a `Presentation`.
+  /// Issuer documents are returned in arbitrary order.
+  ///
+  /// # Errors
+  ///
+  /// Errors if any issuer URL is not a valid `DID` or document resolution fails.
+  pub fn resolve_presentation_issuers(&self, presentation: &WasmPresentation) -> Result<PromiseArrayResolvedDocument> {
+    // TODO: reimplemented function to avoid cloning the entire presentation.
+    //       Would be solved with Rc internal representation, pending memory leak discussions.
+
+    // Extract unique issuers.
+    let mut issuers: HashSet<IotaDID> = HashSet::new();
+    for credential in presentation.0.verifiable_credential.iter() {
+      let issuer: IotaDID = IotaDID::parse(credential.issuer.url().as_str()).wasm_result()?;
+      issuers.insert(issuer);
+    }
```

Comment:
[This part can be simplified in the same way as described in this comment](https://github.com/iotaledger/identity.rs/pull/594/files#r812751853) also applies here.